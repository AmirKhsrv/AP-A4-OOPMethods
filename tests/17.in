credit 101
best_book
best_reviewer
credit 53
recommend_second_approach 121
show_author_info 192
recommend_first_approach 59
best_book
recommend_first_approach 32
recommend_first_approach 22
show_sorted_shelf 2 currently_reading History
recommend_first_approach 28
credit 124
show_sorted_shelf 92 currently_reading Fiction
show_author_info 1237
