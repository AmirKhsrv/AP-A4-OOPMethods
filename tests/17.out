0.014892
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
0.008390
id: 121
Title: Escaping Reality
Genre: Romance
Author: Lisa Renee Jones
Reviews:
id: 227 Rating: 1 Likes: 98 Date: July 19 2019
id: 410 Rating: 3 Likes: 279 Date: February 6 2016
id: 725 Rating: 4 Likes: 0 Date: November 17 2018
id: 1059 Rating: 5 Likes: 0 Date: November 3 2018
id: 1149 Rating: 0 Likes: 0 Date: October 29 2018
id: 1313 Rating: 3 Likes: 1 Date: April 28 2013
id: 1353 Rating: 5 Likes: 3 Date: December 16 2012
id: 1780 Rating: 4 Likes: 4 Date: September 26 2018
id: 1965 Rating: 0 Likes: 0 Date: February 15 2018
id: 2017 Rating: 2 Likes: 1 Date: April 28 2013
id: 2689 Rating: 5 Likes: 0 Date: March 14 2014
id: 2821 Rating: 1 Likes: 0 Date: March 14 2015
id: 4296 Rating: 2 Likes: 1 Date: November 8 2017
id: 4484 Rating: 2 Likes: 0 Date: May 15 2014
id: 4843 Rating: 0 Likes: 0 Date: February 21 2009
id: 4986 Rating: 4 Likes: 0 Date: September 24 2019
id: 5543 Rating: 4 Likes: 0 Date: November 14 2019
id: 6077 Rating: 1 Likes: 0 Date: August 1 2019
id: 6662 Rating: 1 Likes: 2 Date: December 12 2018
id: 7053 Rating: 2 Likes: 14 Date: June 17 2019
id: 7962 Rating: 4 Likes: 0 Date: May 8 2018
id: 9045 Rating: 2 Likes: 0 Date: April 9 2011
id: 9376 Rating: 1 Likes: 0 Date: May 12 2017
id: 9533 Rating: 4 Likes: 0 Date: March 9 2014
id: 10330 Rating: 5 Likes: 0 Date: May 16 2014
id: 10365 Rating: 1 Likes: 0 Date: September 2 2018
id: 10713 Rating: 1 Likes: 16 Date: January 28 2018
id: 11162 Rating: 1 Likes: 0 Date: January 24 2019
id: 11256 Rating: 3 Likes: 0 Date: February 13 2018
id: 11373 Rating: 3 Likes: 0 Date: June 27 2018
id: 11510 Rating: 4 Likes: 0 Date: February 19 2017
id: 12060 Rating: 4 Likes: 0 Date: August 19 2018
id: 12117 Rating: 2 Likes: 5 Date: May 7 2015
id: 12161 Rating: 5 Likes: 0 Date: November 2 2008
id: 12307 Rating: 4 Likes: 0 Date: May 21 2018
id: 14054 Rating: 2 Likes: 0 Date: April 9 2012
id: 14268 Rating: 2 Likes: 1 Date: September 9 2010
id: 14992 Rating: 2 Likes: 0 Date: June 14 2017
id: 15924 Rating: 5 Likes: 1 Date: August 11 2015
id: 16137 Rating: 5 Likes: 0 Date: August 18 2019
id: 16365 Rating: 4 Likes: 0 Date: September 13 2017
id: 16878 Rating: 4 Likes: 0 Date: October 7 2019
id: 192
Name: Marie Force
Year of Birth: 1991
Place of Birth: Armenia
Member Since: February 9 2010
Genres: Romance
Books:
id: 126
Title: Diva
Genre: Historical
Author: Jillian Larkin
Reviews:
id: 139 Rating: 3 Likes: 0 Date: May 2 2019
id: 352 Rating: 5 Likes: 69 Date: January 19 2019
id: 735 Rating: 0 Likes: 0 Date: September 26 2014
id: 986 Rating: 2 Likes: 0 Date: February 2 2010
id: 1590 Rating: 4 Likes: 0 Date: June 16 2014
id: 1922 Rating: 1 Likes: 0 Date: March 14 2014
id: 2169 Rating: 5 Likes: 0 Date: February 13 2019
id: 2589 Rating: 3 Likes: 0 Date: January 24 2017
id: 4273 Rating: 4 Likes: 15 Date: August 7 2015
id: 7605 Rating: 2 Likes: 0 Date: June 29 2019
id: 7689 Rating: 0 Likes: 0 Date: July 3 2018
id: 8101 Rating: 2 Likes: 0 Date: September 18 2019
id: 8195 Rating: 1 Likes: 0 Date: August 11 2015
id: 8542 Rating: 4 Likes: 0 Date: April 21 2019
id: 8727 Rating: 5 Likes: 0 Date: April 4 2012
id: 9078 Rating: 1 Likes: 0 Date: September 10 2010
id: 9288 Rating: 2 Likes: 0 Date: May 26 2019
id: 10364 Rating: 4 Likes: 0 Date: January 27 2014
id: 10655 Rating: 1 Likes: 26 Date: February 3 2015
id: 10780 Rating: 5 Likes: 69 Date: March 25 2018
id: 10864 Rating: 2 Likes: 30 Date: June 19 2014
id: 11013 Rating: 3 Likes: 39 Date: August 21 2019
id: 11044 Rating: 4 Likes: 2 Date: May 17 2019
id: 11592 Rating: 5 Likes: 1 Date: December 17 2017
id: 11745 Rating: 3 Likes: 0 Date: March 14 2013
id: 11830 Rating: 1 Likes: 0 Date: May 3 2010
id: 11840 Rating: 5 Likes: 0 Date: April 27 2014
id: 12010 Rating: 0 Likes: 12 Date: March 16 2014
id: 12491 Rating: 3 Likes: 0 Date: May 3 2018
id: 14089 Rating: 5 Likes: 1 Date: July 4 2011
id: 14123 Rating: 3 Likes: 2 Date: November 8 2018
id: 14318 Rating: 3 Likes: 1 Date: April 12 2018
id: 14999 Rating: 0 Likes: 0 Date: April 8 2011
id: 15041 Rating: 1 Likes: 0 Date: August 9 2009
id: 15364 Rating: 5 Likes: 0 Date: December 29 2018
id: 15486 Rating: 2 Likes: 0 Date: July 17 2017
id: 15570 Rating: 4 Likes: 0 Date: June 22 2019
id: 15906 Rating: 0 Likes: 5 Date: October 7 2017
id: 16287 Rating: 5 Likes: 0 Date: December 17 2019
id: 16779 Rating: 5 Likes: 0 Date: May 7 2014
id: 16961 Rating: 0 Likes: 0 Date: October 21 2018
id: 17016 Rating: 5 Likes: 0 Date: July 25 2018
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 41
Title: If You Find Me
Genre: Young Adult
Author: Emily Murdoch
Reviews:
id: 561 Rating: 2 Likes: 0 Date: September 24 2015
id: 1744 Rating: 2 Likes: 0 Date: August 22 2013
id: 2137 Rating: 3 Likes: 0 Date: July 29 2019
id: 2170 Rating: 0 Likes: 2 Date: August 18 2013
id: 2179 Rating: 2 Likes: 0 Date: December 3 2019
id: 2650 Rating: 5 Likes: 0 Date: May 1 2015
id: 2770 Rating: 2 Likes: 0 Date: April 22 2019
id: 2950 Rating: 2 Likes: 0 Date: August 23 2018
id: 3308 Rating: 4 Likes: 0 Date: June 29 2018
id: 3795 Rating: 0 Likes: 0 Date: July 14 2013
id: 3960 Rating: 2 Likes: 0 Date: May 26 2016
id: 4745 Rating: 5 Likes: 0 Date: May 6 2015
id: 4778 Rating: 1 Likes: 0 Date: December 6 2019
id: 4846 Rating: 4 Likes: 0 Date: November 27 2017
id: 5011 Rating: 4 Likes: 0 Date: April 4 2017
id: 5460 Rating: 5 Likes: 0 Date: November 2 2017
id: 5880 Rating: 4 Likes: 0 Date: May 1 2018
id: 6912 Rating: 3 Likes: 0 Date: February 7 2019
id: 7550 Rating: 2 Likes: 1 Date: July 10 2013
id: 7818 Rating: 3 Likes: 0 Date: July 23 2016
id: 7840 Rating: 5 Likes: 0 Date: February 21 2018
id: 7843 Rating: 5 Likes: 0 Date: March 10 2015
id: 7893 Rating: 2 Likes: 0 Date: October 13 2018
id: 8388 Rating: 1 Likes: 0 Date: April 19 2011
id: 8543 Rating: 1 Likes: 0 Date: March 16 2019
id: 8961 Rating: 2 Likes: 0 Date: December 5 2016
id: 9559 Rating: 5 Likes: 0 Date: May 21 2019
id: 9653 Rating: 2 Likes: 0 Date: November 10 2014
id: 9903 Rating: 1 Likes: 0 Date: March 26 2015
id: 11654 Rating: 3 Likes: 1 Date: December 17 2013
id: 12221 Rating: 0 Likes: 0 Date: August 16 2018
id: 12674 Rating: 2 Likes: 0 Date: October 19 2019
id: 13086 Rating: 0 Likes: 0 Date: June 1 2019
id: 13204 Rating: 2 Likes: 0 Date: April 1 2013
id: 13301 Rating: 1 Likes: 0 Date: August 26 2018
id: 13724 Rating: 4 Likes: 0 Date: July 19 2017
id: 14310 Rating: 5 Likes: 0 Date: November 3 2012
id: 14798 Rating: 5 Likes: 0 Date: September 2 2019
id: 14855 Rating: 1 Likes: 0 Date: April 11 2016
id: 15094 Rating: 2 Likes: 0 Date: October 14 2019
id: 15656 Rating: 3 Likes: 0 Date: July 12 2019
id: 15954 Rating: 1 Likes: 0 Date: May 28 2019
id: 16170 Rating: 5 Likes: 0 Date: December 2 2010
id: 16470 Rating: 4 Likes: 0 Date: November 26 2010
id: 16757 Rating: 5 Likes: 0 Date: June 6 2018
id: 16758 Rating: 4 Likes: 0 Date: November 1 2014
id: 16988 Rating: 0 Likes: 0 Date: May 19 2018
id: 17066 Rating: 5 Likes: 0 Date: November 22 2019
id: 392
Title: From the Mixed-Up Files of Mrs. Basil E. Frankweiler
Genre: Fiction
Author: E.L. Konigsburg
Reviews:
id: 356 Rating: 3 Likes: 0 Date: August 24 2018
id: 396 Rating: 2 Likes: 558 Date: January 16 2011
id: 677 Rating: 5 Likes: 3 Date: August 13 2018
id: 914 Rating: 5 Likes: 0 Date: June 3 2019
id: 1654 Rating: 0 Likes: 0 Date: February 6 2019
id: 1665 Rating: 5 Likes: 0 Date: November 21 2017
id: 1866 Rating: 0 Likes: 0 Date: December 27 2018
id: 2173 Rating: 2 Likes: 0 Date: February 17 2012
id: 2285 Rating: 5 Likes: 0 Date: March 28 2013
id: 2901 Rating: 5 Likes: 0 Date: July 20 2017
id: 4481 Rating: 5 Likes: 2 Date: May 21 2016
id: 5827 Rating: 4 Likes: 0 Date: November 25 2019
id: 5975 Rating: 1 Likes: 0 Date: March 28 2018
id: 6899 Rating: 5 Likes: 1 Date: February 27 2018
id: 8914 Rating: 1 Likes: 0 Date: July 12 2013
id: 9248 Rating: 5 Likes: 0 Date: October 4 2018
id: 9999 Rating: 5 Likes: 0 Date: March 18 2015
id: 10317 Rating: 4 Likes: 0 Date: April 18 2014
id: 10521 Rating: 4 Likes: 0 Date: September 13 2019
id: 10822 Rating: 5 Likes: 0 Date: May 11 2016
id: 11689 Rating: 5 Likes: 1 Date: January 27 2017
id: 12274 Rating: 5 Likes: 0 Date: August 25 2015
id: 12379 Rating: 5 Likes: 0 Date: June 23 2015
id: 12431 Rating: 0 Likes: 1 Date: July 25 2019
id: 12626 Rating: 3 Likes: 0 Date: September 18 2019
id: 13444 Rating: 0 Likes: 0 Date: June 26 2013
id: 13813 Rating: 3 Likes: 0 Date: May 27 2018
id: 14083 Rating: 3 Likes: 0 Date: October 26 2014
id: 14142 Rating: 4 Likes: 0 Date: December 27 2014
id: 14199 Rating: 3 Likes: 4 Date: October 12 2015
id: 14427 Rating: 1 Likes: 2 Date: June 15 2018
id: 16431 Rating: 5 Likes: 0 Date: October 28 2014
id: 16509 Rating: 5 Likes: 0 Date: March 17 2016
id: 16740 Rating: 3 Likes: 0 Date: April 26 2016
id: 16829 Rating: 0 Likes: 0 Date: October 15 2017
id: 17340 Rating: 3 Likes: 0 Date: November 13 2017
id: 42
Title: Tigers and Devils
Genre: Romance
Author: Sean Kennedy
***
id: 1
Title: Forging Zero
Genre: Science Fiction
Author: Sara  King
Reviews:
id: 416 Rating: 2 Likes: 1 Date: April 4 2014
id: 708 Rating: 4 Likes: 0 Date: October 19 2010
id: 866 Rating: 0 Likes: 0 Date: July 5 2016
id: 889 Rating: 1 Likes: 0 Date: December 21 2017
id: 1572 Rating: 2 Likes: 0 Date: April 18 2015
id: 1766 Rating: 2 Likes: 0 Date: June 8 2013
id: 2152 Rating: 1 Likes: 3 Date: January 2 2019
id: 2948 Rating: 1 Likes: 0 Date: October 14 2017
id: 3017 Rating: 4 Likes: 0 Date: July 25 2018
id: 4267 Rating: 0 Likes: 1 Date: January 14 2015
id: 4640 Rating: 1 Likes: 0 Date: September 9 2016
id: 5212 Rating: 4 Likes: 0 Date: January 12 2010
id: 5554 Rating: 5 Likes: 0 Date: January 27 2015
id: 6186 Rating: 0 Likes: 0 Date: July 14 2018
id: 6356 Rating: 5 Likes: 3 Date: December 13 2015
id: 6574 Rating: 4 Likes: 9 Date: March 15 2018
id: 7151 Rating: 3 Likes: 5 Date: July 18 2014
id: 7884 Rating: 3 Likes: 0 Date: August 29 2013
id: 9223 Rating: 3 Likes: 0 Date: October 20 2011
id: 9810 Rating: 2 Likes: 0 Date: June 5 2016
id: 11056 Rating: 2 Likes: 0 Date: October 16 2012
id: 11581 Rating: 4 Likes: 4 Date: July 24 2015
id: 11997 Rating: 5 Likes: 3 Date: November 19 2015
id: 12488 Rating: 4 Likes: 0 Date: September 17 2019
id: 12559 Rating: 5 Likes: 0 Date: March 5 2010
id: 13030 Rating: 2 Likes: 2 Date: November 23 2017
id: 13083 Rating: 2 Likes: 0 Date: August 8 2019
id: 13266 Rating: 2 Likes: 0 Date: January 7 2015
id: 13897 Rating: 1 Likes: 0 Date: February 28 2014
id: 15083 Rating: 5 Likes: 0 Date: July 24 2018
id: 16304 Rating: 0 Likes: 0 Date: August 10 2018
0.010959
id: 80
Title: We Were the Mulvaneys
Genre: Fiction
Author: Joyce Carol Oates
***
id: 254
Title: Caps for Sale: A Tale of a Peddler Some Monkeys and Their Monkey Business
Genre: Childrens
Author: Esphyr Slobodkina
***
id: 107
Title: Finger Lickin' Fifteen
Genre: Mystery
Author: Janet Evanovich
***
id: 231
Title: Forbidden Attraction
Genre: Romance
Author: S.R. Watson
***
id: 331
Title: Golden Trail
Genre: Romance
Author: Kristen Ashley
***
id: 236
Title: Slow Burn
Genre: Romance
Author: Sam B. Morgan
***
id: 1237
Name: Laura Joffe Numeroff
Year of Birth: 1993
Place of Birth: Tunisia
Member Since: May 4 2013
Genres: Children S Books
Books:
